extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -422,15 +457,23 @@ export class AmpA4A extends AMP.BaseElement {
       if (rendered instanceof Error) {
         // If we got as far as getting a URL, then load the ad, but note the
         // error.
-        if (this.adUrl_) {
+        if (this.shouldRenderViaSafeFrame_ == 'safeframe' &&
```

Comment:
Not a big fan of the code duplication here.  I also think we may have a bug given that we throw rendered prior to setting this.rendered_ = true (though I cannot think of what code references this.rendered_ after layoutcallback execution).  Let's rework this as follows:

if (rendered instanceof Error || !rendered) {
    // render code
}
this.rendered_ = true;
if (rendered instanceof Error) {
    throw rendered;
}